viam-orb-slam3.go

```diff
@@ -489,11 +489,25 @@ func (orbSvc *orbslamService) GetSLAMProcessConfig() pexec.ProcessConfig {
 	args = append(args, "-port="+orbSvc.port)
 	args = append(args, "--aix-auto-update")
 
+	target := orbSvc.executableName
+
+	appDir := os.Getenv("APPDIR")
+
+	if appDir != "" {
+		// The orb grpc server is expected to be in
+		// /usr/bin/ if we are running in an appimage.
+		target = appDir + "/usr/bin/" + strings.TrimPrefix(target, "/")
+	}
+
 	return pexec.ProcessConfig{
-		ID:      "slam_orbslamv3",
-		Name:    orbSvc.executableName,
-		Args:    args,
-		Log:     true,
+		ID:   "slam_orbslamv3",
+		Name: target,
+		Args: args,
+		Log:  true,
+		// In appimage this is set to the appimage
+		// squashfs mount location (/tmp/.mountXXXXX)
+		// Otherwise, it is an empty string
+		CWD:     os.Getenv("APPRUN_RUNTIME"),
```

Comment:
Could we confirm we haven't regressed the behavior of running orbslam outside of an appimage?

Reply:
@EshaMaharishi 
We are not able to run the orb_carto_grpc server on linux outside of an appimage. This appears (I'm not a ldd expert) to be due to the fact that the `libORB_SLAM3.so` is bing linked to the build directory of the ORB_SLAM3 fork, which is not present if we copy & paste the orb_grpc_server binary outside of an app image.

Within canon:
```
testbot@049f063a8033:/host$ ldd viam-orb-slam3/bin/orb_grpc_server
        linux-vdso.so.1 (0x0000ffffa8f14000)
... bunch of deps
        libORB_SLAM3.so => /host/viam-orb-slam3/ORB_SLAM3/lib/libORB_SLAM3.so (0x0000ffffa8423000)
... bunch more deps
```

Within canon:
```
testbot@049f063a8033:/host$ ldd viam-orb-slam3/bin/orb_grpc_server
        linux-vdso.so.1 (0x0000ffffa8f14000)
... bunch of deps
        libORB_SLAM3.so => /host/viam-orb-slam3/ORB_SLAM3/lib/libORB_SLAM3.so (0x0000ffffa8423000)
... bunch more deps
```

Same binary on the pi:
```
nickpi@nick-slam-dev-pi:~ $ ldd /usr/local/bin/orb_grpc_server
        linux-vdso.so.1 (0x0000007fb5374000)
... bunch of deps
        libORB_SLAM3.so => not found
... bunch more deps
```

This is the same behavior I get on viam-orb-slam3 main, so this isn't new behavior.

May I still move forward with merging this change?

Reply:
Ah, interesting. Yes, I think we can merge given that it works within the appimage.

Reply:
> We are not able to run the orb_carto_grpc server (*orb_grpc_server*) on linux outside of an appimage.

I don't think this is correct. The following invocation worked outside of an appimage and should still work.
`make clean bufinstall buf setup build install`
Then running the code as a module will correctly find orb_grpc_server (which is installed to the system).

Reply:
@zaporter-work I think that only works if the build system is the same as the target system. 

If you just copy `orb_grpc_server` from the build system it doesn't work as it is looking for a dynamic library back on the build system which doesn't exist on the target system.

Reply:
I agree with that, but you can still run it outside of an appimage, you just have to build and install it locally on the device. (or undergo a more complicated install process)

Reply:
I've updated my comment to reflect that nuance.